src/parser/jobs/blm/BlmCooldowns.js

```diff
@@ -0,0 +1,148 @@
+//import React, {Fragment} from 'react'
+//import {ActionLink} from 'components/ui/DbLink'
+import ACTIONS, {getAction} from 'data/ACTIONS'
+//import STATUSES from 'data/STATUSES'
+import Module from 'parser/core/Module'
+//import {Suggestion, SEVERITY} from 'parser/core/modules/Suggestions'
+import {Rule, Requirement} from 'parser/core/modules/Checklist'
+
+//tracking the important™ CDs
+const TRACKEDCDS = [
+	ACTIONS.LEY_LINES.id,
+	ACTIONS.SHARPCAST.id,
+	ACTIONS.TRIPLECAST.id,
+	ACTIONS.CONVERT.id,
+]
+
+//Time that laqi deems ok for a OGCD to be down : ^)
+const DOWNTIME_OK_TIME = 15000
+
+export default class Blmcooldowns extends Module {
+	static handle = 'blmcooldowns'
+	static dependencies = [
+		'combatants',
+		'cooldowns',
+		'suggestions',
+		'gauge',
+		'checklist',
+	]
+
+	_downTime = {}
+
+	constructor(...args) {
+		super(...args)
+		this.addHook('begincast', {by: 'player'}, this._onBegin)
+		this.addHook('cast', {by: 'player'}, this._onCast)
+		this.addHook('complete', this._onComplete)
+	}
+
+	_onBegin(event) {
+
+		//do the check on begin (even tho they don't have a cast time)
+		this._cooldownCheck(event)
+	}
+
+	_onCast(event) {
+
+		//do the check at every cast
+		this._cooldownCheck(event)
+	}
+
+	_onComplete(event) {
+		const endTime = event.timestamp
+		const startTime = this.parser.fight.start_time
+		const encounterLength = endTime - startTime
+		const OGCDRequirements = []
+		Object.keys(this._downTime).forEach(id => {
+			const dt = this._downTime[id]
+
+			//wrap up all the open ones and save them in history
+			if (dt.current) {
+				dt.current.stoptime = endTime
+				dt.history.push(dt.current)
+				dt.current = null
+			}
+
+			//calculate the downtime based on the start and stop values and sum the array
+			const totalSumOfDownTime = dt.history.map(downTime => {
+				return Math.max(downTime.stoptime - downTime.starttime, 0)
+			}).reduce(
+				(accumulator, currentValue) => accumulator + currentValue
+			)
+
+			//write the results as a new Requirement to show up later
+			OGCDRequirements.push(
+				new Requirement({
+					name: getAction(id).name,
+					percent: this._percentFunction(id, totalSumOfDownTime, encounterLength),
+				})
+			)
+		})
+
+		//new Rule and adds the array of Requirements that just got generated
+		this.checklist.add(new Rule({
+			name: 'Use your OGCDs',
+			description: 'Always make sure to use your OGCDs when they are up but don\'t clip them. Utilize your procs or fast Blizzard III or Fire IIIs.',
+			requirements: OGCDRequirements,
+		}))
+	}
+
+	//cool function that I invented that just sets ok usage as 98% and falls very quickly
+	_percentFunction(actionId, downtime, fightlength) {
+		const cooldown = getAction(actionId).cooldown
+		const numberOfUses = Math.floor(fightlength/(cooldown*1000))
+		return Math.exp(-downtime*0.03/(DOWNTIME_OK_TIME*numberOfUses))*100
```

Comment:
As before, I'm concerned about how opaque this is. The end-user will probably think it's bugging out or something, and possibly ignore it.

Reply:
The function itself isn't opaque at all. But all it does is:
```
Downtime: 0s => 100%
Downtime: ALLOWEDOGCDDOWNTIME * MAXPOSSIBLENUMBEROFUSES => 98%
Downtime: way more than that => low%
```
above 95% is green. I've tested various logs from different percentages.
Good people stay easily at ~98%. People around blue/purple get their 95%+.
I would give it a try and if I hear from people who use it "it's confusing" "I feel like it's worthless" etc. then I would get rid of it or replace it with something else.